htsget-http-actix/src/main.rs

```diff
@@ -0,0 +1,47 @@
+use actix_web::{web, App, HttpServer};
+use htsget_search::{htsget::from_storage::HtsGetFromStorage, storage::local::LocalStorage};
+use std::env::args;
+mod config;
+use config::Config;
+mod get;
+mod post;
+
+const USAGE: &str = r"
+This executable doesn't use command line arguments, but there are some environment variables that can be set to configure the HtsGet server:
+* HTSGET_IP: The ip to use. Default: 127.0.0.1
+* HTSGET_PORT: The port to use. Default: 8080
+* HTSGET_PATH: The path to the directory where the server should be started. Default: Actual directory
+";
+
+type HtsGetStorage = HtsGetFromStorage<LocalStorage>;
+#[actix_web::main]
+async fn main() -> std::io::Result<()> {
+  if args().len() > 1 {
+    // Show help if command line options are provided
+    println!("{}", USAGE);
+    return Ok(());
+  }
+  let config = envy::from_env::<Config>().expect("The environment variables weren't properly set!");
+  let htsget = HtsGetFromStorage::new(
+    LocalStorage::new(config.htsget_path)
+      .expect("Couldn't create a Storage with the provided path"),
+  );
+  let htsget = web::Data::new(htsget);
+  HttpServer::new(move || {
+    App::new()
+      .app_data(htsget.clone())
```

Comment:
I would rather create a different instance of HtsGet per thread, rather than sharing the reference across threads.
Given that htsget is stateless the current approach should not give problems, but if at any moment we need to keep state (for watever reason, think in the storage needing some cache for example), it might give trouble.

In summary, I would create a different instance of htsget (or clone) within this closure, then I would use `.data()` rather than `.app_data()`.

Reply:
Yeah, that makes sense!